lib/block_view_derived_key_test.go

```diff
@@ -18,6 +20,100 @@ const (
+		chain.AddInputsAndChangeToTransaction(txn, 10, mempool)
+	require.NoError(err)
+	require.Equal(totalInput, spendAmount+changeAmount+fees)
+	require.Greater(totalInput, uint64(0))
+
+	if isSignerSender {
+		// Sign the transaction with the provided derived key
+		_signTxn(t, txn, signerPriv)
+	} else {
+		// Sign the transaction with the provided derived key
+		_signTxnWithDerivedKey(t, txn, signerPriv)
+	}
+
+	// Get utxoView if it doesn't exist
+	if mempool != nil {
+		utxoView, err = mempool.GetAugmentedUniversalView()
+		require.NoError(err)
+	}
+	if utxoView == nil {
+		utxoView, err = NewUtxoView(db, params, chain.postgres, chain.snapshot)
+		require.NoError(err)
+	}
+
+	txHash := txn.Hash()
+	blockHeight := chain.blockTip().Height + 1
+	utxoOps, _, _, _, err :=
+		utxoView.ConnectTransaction(txn, txHash, getTxnSize(*txn), blockHeight,
+			true /*verifySignature*/, false /*ignoreUtxos*/)
+	return utxoOps, txn, err
+}
+
+// Verify that the balance and expiration block in the db match expectation.
+func _derivedKeyVerifyTest(t *testing.T, db *badger.DB, chain *Blockchain, transactionSpendingLimit *TransactionSpendingLimit,
+	derivedPublicKey []byte, expirationBlockExpected uint64, balanceExpected uint64,
+	operationTypeExpected AuthorizeDerivedKeyOperationType, mempool *DeSoMempool) {
+
+	require := require.New(t)
+	_ = require
+
+	senderPkBytes, _, err := Base58CheckDecode(senderPkString)
+	require.NoError(err)
+
+	// Verify that expiration block was persisted in the db or is in mempool utxoView
+	if mempool == nil {
+		derivedKeyEntry := NewDbAdapter(chain).GetOwnerToDerivedKeyMapping(*NewPublicKey(senderPkBytes), *NewPublicKey(derivedPublicKey))
+		// If we removed the derivedKeyEntry from utxoView altogether, it'll be nil.
+		// To pass the tests, we initialize it to a default struct.
+		if derivedKeyEntry == nil || derivedKeyEntry.isDeleted {
+			derivedKeyEntry = &DerivedKeyEntry{
+				*NewPublicKey(senderPkBytes), *NewPublicKey(derivedPublicKey), 0, AuthorizeDerivedKeyOperationValid, nil, transactionSpendingLimit, nil, false}
+		}
+		require.Equal(derivedKeyEntry.ExpirationBlock, expirationBlockExpected)
+		require.Equal(derivedKeyEntry.OperationType, operationTypeExpected)
+	} else {
+		utxoView, err := mempool.GetAugmentedUniversalView()
+		require.NoError(err)
+		derivedKeyEntry := utxoView._getDerivedKeyMappingForOwner(senderPkBytes, derivedPublicKey)
+		// If we removed the derivedKeyEntry from utxoView altogether, it'll be nil.
+		// To pass the tests, we initialize it to a default struct.
+		if derivedKeyEntry == nil || derivedKeyEntry.isDeleted {
+			derivedKeyEntry = &DerivedKeyEntry{*NewPublicKey(senderPkBytes), *NewPublicKey(derivedPublicKey), 0, AuthorizeDerivedKeyOperationValid, nil, transactionSpendingLimit, nil, false}
+		}
+		require.Equal(derivedKeyEntry.ExpirationBlock, expirationBlockExpected)
+		require.Equal(derivedKeyEntry.OperationType, operationTypeExpected)
+	}
```

Comment:
```suggestion
	var derivedKeyEntry *DerivedKeyEntry
	// Verify that expiration block was persisted in the db or is in mempool utxoView
	if mempool == nil {
		derivedKeyEntry := NewDbAdapter(chain).GetOwnerToDerivedKeyMapping(*NewPublicKey(senderPkBytes), *NewPublicKey(derivedPublicKey))
	} else {
		utxoView, err := mempool.GetAugmentedUniversalView()
		require.NoError(err)
		derivedKeyEntry := utxoView._getDerivedKeyMappingForOwner(senderPkBytes, derivedPublicKey)
	}
	// If we removed the derivedKeyEntry from utxoView altogether, it'll be nil.
	// To pass the tests, we initialize it to a default struct.
	if derivedKeyEntry == nil || derivedKeyEntry.isDeleted {
		derivedKeyEntry = &DerivedKeyEntry{*NewPublicKey(senderPkBytes), *NewPublicKey(derivedPublicKey), 0, AuthorizeDerivedKeyOperationValid, nil, transactionSpendingLimit, nil, false}
	}
	require.Equal(derivedKeyEntry.ExpirationBlock, expirationBlockExpected)
	require.Equal(derivedKeyEntry.OperationType, operationTypeExpected)
```
we can de-dupe this code here.